File: Source/Fayde.Server/Media/VSM/VisualState.cs
using Fayde.Core;
using Fayde.Media.Animation;
using Fayde.Xaml.Metadata;

namespace Fayde.Media.VSM
{
    public class VisualState : DependencyObject
    {
        public static readonly PropertyDescription Storyboard = PropertyDescription.Register("Storyboard", typeof(Storyboard), typeof(VisualState), true);
    }
}

File: Source/Fayde.Client/Typescript/Media/VSM/VisualState.ts
/// <reference path="../../Core/DependencyObject.ts" />
/// <reference path="../../Core/XamlObjectCollection.ts" />
/// CODE
/// <reference path="../Animation/Storyboard.ts" />

module Fayde.Media.VSM {
    export class VisualState extends DependencyObject {
        static StoryboardProperty: DependencyProperty = DependencyProperty.Register("Storyboard", () => Animation.Storyboard, VisualState);
        Storyboard: Animation.Storyboard;
        static Annotations = { ContentProperty: VisualState.StoryboardProperty };
    }
    Fayde.RegisterType(VisualState, {
    	Name: "VisualState",
    	Namespace: "Fayde.Media.VSM",
    	XmlNamespace: Fayde.XMLNS
    });

    export class VisualStateCollection extends XamlObjectCollection<VisualState> {
    }
    Fayde.RegisterType(VisualStateCollection, {
    	Name: "VisualStateCollection",
    	Namespace: "Fayde.Media.VSM",
    	XmlNamespace: Fayde.XMLNS
    });
}